### ISSUE TITLE
Set up cron job with actions to compile automatically

### ISSUE THREAD

**Issue by FichteFoll** [SublimeText/sublime-text-docset#2]
And a feed, so Dash and Zeal can update them within the app.

**Comment by FichteFoll** [comment]
Problem: The contents of the website are copyrighted and unless we are granted a license, we aren't allowed to redistribute. Will need to ask Jon about this directly.

**Comment by michaelblyons** [comment]
> Problem: The contents of the website are copyrighted and unless we are granted a license, we aren't allowed to redistribute. Will need to ask Jon about this directly.

Resolved in #10

**Comment by michaelblyons** [comment]
How do you imagine this works? GitHub Actions has

```yaml
on:
 schedule:
   - cron:  "42 7 15 * *"
```

We could have it spin up once a day/week/etc., check @maliayas's [documentation diff-er](https://github.com/maliayas/SublimeText_Documentation) ([Atom feed](https://github.com/maliayas/SublimeText_Documentation/commits/master.atom)) to see whether there's something new, spool up a VM, and build the docset.

**Comment by FichteFoll** [comment]
I believe a regular schedule should be fine, with some diff checks between the previous run. With that, a run every day or so would be fast enough to be updated along with a new release without putting any noticable load on the website.

I'm unsure if we need human review in form of a PR or similar before releasing it to the public, though. The review process has the advantage of not publishing a broken version in case the docs are updated in a majorly incompatible way, but on the other hand that requires more work to set up and review (which could be a bottleneck). Generally, I doubt we will run into breakage a lot and would rather clean up any potential mess after it happens for this project.

(That's of course assuming that we get into the Dash/Zeal feed for auto-updates.)

**Comment by michaelblyons** [comment]
Maybe it could generate a pre-release automatically, and you promote it (if possible) or re-release it (if not) as a stable version after review.

**Comment by FichteFoll** [comment]
I would then rather prefer the PR with a diff and then automatically creating a release from `master`.

**Comment by michaelblyons** [comment]
But there would be no code to put in the PR. Maybe open an issue whose text include the diff between old docs and new docs? Then you can build locally, verify that the new items are indexed (if not, _then_ make a PR to merge), and make a release?

**Comment by FichteFoll** [comment]
Right, not in the default branch anyway. We could use a separate branch, though, think `gh-branches`, that's entirely separate from the code and contains only the built version which will be used for distribution. That would be reviewable and diffable.

I just don't believe fetching a built docset and comparing it locally is the way to go here (or even generate a diff otherwise because then the version to diff it with would need to be stored somewhere, too, in which case we might as well go the PR way).

**Comment by maliayas** [comment]
In case it's useful for anyone, here is the command I use on https://github.com/maliayas/SublimeText_Documentation

```
wget \
    --mirror \
    --no-parent \
    --page-requisites \
    --convert-links \
    --continue \
    --adjust-extension \
    --user-agent="" \
    --execute="robots=off" \
    --wait=1 \
    --append-output="mirror.log" \
    --no-verbose \
    --max-redirect=0 \
    https://www.sublimetext.com/docs/index.html
```

And I want to add that, this process requires a review. Sometimes network fails, sometimes ST web site fails, so every now and then unintentional commits occur. I locally undo them on my PC and never push to GH. On the other hand, I try to separate trivial and serious changes into separate commits and only git-tag the latter (also a review process).

P.S. Thanks @kapitanluffy for the ping. I've just found your email in my spam box. Since my email is sort of broken, I'm replying here.

Cheers!

**Comment by michaelblyons** [comment]
Closing in favor of #39 since the scope here kind of sprawled, and most if not all of it is now finished. If there's more functionality needed, let's open a new issue and link back here.

**Comment by FichteFoll** [comment]
Indeed, since we have dependabot notfying us about changes to our submodules (#18), that part is already taken care of. Since each change will thus also cause a new commit, we can then use GitHub actions to more or less automatically trigger a build of the docsets and just need something to update the feed with them, which is tracked in #39.
